### PR TITLE
[SDTEST-2055] Add documentation for AI flaky test categorization

### PR DESCRIPTION
**Pull request by ManuelPalenzuelaDD**
### What does this PR do? What is the motivation?
We've recently released a new feature for the Flaky Test Management product which categorizes your flaky tests with AI. We want to document its behavior.

### Merge instructions

Merge readiness:
- [x] Ready for merge

### CODE REVIEW

**Review by TovRudyy — APPROVED** [review]

**Review by daniel-mohedano — APPROVED** [review]

**Comment by joepeeples** [comment]
Opened [DOCS-11573](https://datadoghq.atlassian.net/browse/DOCS-11573) and assigned myself to follow up with editorial review.

[DOCS-11573]: https://datadoghq.atlassian.net/browse/DOCS-11573?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

File: content/en/tests/flaky_management/_index.md [hunk 1]
```diff
@@ -73,6 +74,31 @@ When you fix a flaky test, Test Optimization's remediation flow can confirm the
    - If all retries pass, updates the test's status to `Fixed`.
    - If any retry fails, keeps the test's current status (`Active`, `Quarantined`, or `Disabled`).
 
+## AI-powered flaky test categorization
+
+Flaky Test Management uses AI to automatically assign a root cause category to each flaky test based on execution patterns and error signals. This helps you filter, triage, and prioritize flaky tests more effectively.
+
+<div class="alert alert-info"><strong>Note:</strong> A test must have at least one failed execution that includes both <code>@error.message</code> and <code>@error.stack</code> tags to be eligible for categorization. If the test was recently detected, categorization may take some time to complete.</div>
```
Comment (joepeeples) [inline content/en/tests/flaky_management/_index.md]: Is it possible to give a more specific time estimate? Something like "may take several minutes/hours" or "up to x minutes" so customers know better what to expect.

File: content/en/tests/flaky_management/_index.md [hunk 2]
```diff
@@ -73,6 +74,31 @@ When you fix a flaky test, Test Optimization's remediation flow can confirm the
...
+| **Concurrency** | Failures from unsafe thread interactions or race conditions. |
+| **Randomness** | Relies on random values; edge cases may trigger failures. |
+| **Floating Point** | Precision issues or rounding errors affect comparisons. |
+| **Unordered Collection** | Assumes a fixed iteration order for unordered data. |
+| **Too Restrictive Range** | Assertions miss valid edge-case outputs. |
+| **Timeout** | Exceeds time limits due to variable execution time. |
+| **Order Dependency** | Relies on shared state from other tests. |
+| **Resource Leak** | Does not release memory, files, or sockets. |
+| **Asynchronous Wait** | Uses fixed delays instead of awaiting completion. |
+| **IO** | Impacted by disk, file, or device conditions. |
```
Comment (joepeeples) [inline content/en/tests/flaky_management/_index.md]: Our style linter suggests `I/O` instead of `IO`, but we should use whichever format appears in the UI.

Comment (ManuelPalenzuelaDD) [reply content/en/tests/flaky_management/_index.md]: In the UI we show this as `IO` 👍

File: content/en/tests/flaky_management/_index.md (hunk 1 above)
Comment (ManuelPalenzuelaDD) [reply content/en/tests/flaky_management/_index.md]: Agreed 👍
